MelodyGenerator.js

```diff
@@ -0,0 +1,145 @@
+const durations = require('./durations.js');
+const MelodyQueue = require('./MelodyQueue.js');
+const waveGenerator = require('./WaveGenerator.js');
+const note = require('./note.js');
+
+//Global variables that are accessible to all
+let melody = new MelodyQueue();
+let macros = new Map(); // In case a macro needs to be defined
+const regex1 = /repeat (\d) times (.+)/; // Regex for repeat commands
```

Comment:
to parse DSL or another computer languages is nice to split into tokens. Tokens can be parsed with regexs, but regexs cannot parse whole code (because code is not a *regular* expression).

Regex's syntax is very hard but If will you learn then regexs be very powerfull - like a magic ;) https://xkcd.com/208/

Nice examples when regexs are wrong:
1. HTML: https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags
2. Email valid: https://emailregex.com/